backend/usage_v2/helper.py

```
@@ -65,6 +65,45 @@ def get_aggregated_token_count(run_id: str) -> dict:
             logger.error(f"An unexpected error occurred for run_id {run_id}: {str(e)}")
             raise APIException("Error while aggregating token counts")
 
+    @staticmethod
+    def get_aggregated_cost(execution_id: str) -> Optional[float]:
+        """Retrieve aggregated cost for the given execution_id.
+
+        Args:
+            execution_id (str): The identifier for the
+            total cost of a particular execution.
+
+        Returns:
+        Optional[float]: The total cost in dollars if available, else None.
+
+        Raises:
+            APIException: For unexpected errors during database operations.
+        """
+        try:
+            # Aggregate the cost for the given execution_id
+            total_cost = Usage.objects.filter(execution_id=execution_id).aggregate(
+                cost_in_dollars=Sum(UsageKeys.COST_IN_DOLLARS)
+            )[UsageKeys.COST_IN_DOLLARS]
+
+            logger.debug(
+                f"Cost aggregated successfully for execution_id: {execution_id}"
+            )
+
+            return total_cost
+
+        except Usage.DoesNotExist:
+            # Handle the case where no usage data is found for the given execution_id
+            logger.warning(
+                f"Usage data not found for the specified execution_id: {execution_id}"
+            )
+            return None
+        except Exception as e:
+            # Handle any other exceptions that might occur during the execution
+            logger.error(
+                f"An unexpected error occurred for execution {execution_id}: {str(e)}"
+            )
+            raise APIException("Error while aggregating cost")
```

@pk-zipstack in the name of exception handling we should'nt catch and re-raise a different error. While catching errors we should try to be specific about the type we wish to handle. Catching all pythonic errors this way is an antipattern (depends on the usecase)

We should ideally allow the error to propagate up the call stack so that all necessary information about the error is preserved. Since this cost aggregation is not a critical step that needs to be evaluated we can choose to suppress the exception here - in that regard I suggest this
1. Log a warning indicating that there's something wrong with the cost aggregation (add info about the error as well, consider a traceback if we call this function from multiple places)
1. Return a `None` instead of re-raising the error
2. The caller of this function needs to handle appropriately based on the `None`